File: Symfony/src/BGR/Serrano/ProductoBundle/Entity/Proveedor.php
<?php

namespace BGR\Serrano\ProductoBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use JMS\Serializer\Annotation\Type;
use JMS\Serializer\Annotation\Exclude;

/**
 * Proveedor
 *
 * @ORM\Table()
 * @ORM\Entity(repositoryClass="BGR\Serrano\ProductoBundle\Entity\ProveedorRepository")
 */
class Proveedor
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @Type("integer")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="name", type="string", length=255)
     * @Type("string")
     */
    private $name;

    /**
     * @ORM\ManyToMany(targetEntity="Producto", mappedBy="unidad_de_medidas")
     * @Type("ArrayCollection<BGR\Serrano\ProductoBundle\Entity\Producto>")
     * @Exclude
     */
    private $productos;
    
    /**
     * Get id
     *
     * @return integer 
     */
    public function getId()
    {
        return $this->id;
        
    }

    /**
     * Set name
     *
     * @param string $name
     * @return Proveedor
     */
    public function setName($name)
    {
        $this->name = $name;
    
        return $this;
    }

    /**
     * Get name
     *
     * @return string 
     */
    public function getName()
    {
        return $this->name;
    }
	public function getProductos() {
		return $this->productos;
	}
	public function setProductos($productos) {
		$this->productos = $productos;
		return $this;
	}
	
}

File: BGR/BGR-Data/src/main/java/ar/com/bgr/serrano/model/Proveedor.java
/**
 * 
 */
package ar.com.bgr.serrano.model;

import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import org.codehaus.jackson.annotate.JsonIgnore;

/**
 * 
 * Descripcion:
 * 
 * {}
 *
 * @author matias
 * 
 * @since 01/06/2014
 */
@Entity(name="Proveedor")
@Table(name="Proveedor")
public class Proveedor {

	@Id
	@GeneratedValue
	private Integer id;
	
	@Column(name="name")
	private String name;		
		
	@JsonIgnore
	@OneToMany(fetch = FetchType.EAGER, mappedBy = "productoProveedoresId.proveedor", cascade=CascadeType.ALL)
	private Set<ProductoProveedor> productoProveedoresId;
	
	/**
	 * @return the id
	 */
	public Integer getId() {
		return id;
	}

	/**
	 * @param id the id to set
	 */
	public void setId(Integer id) {
		this.id = id;
	}

	/**
	 * @return the name
	 */
	public String getName() {
		return name;
	}

	/**
	 * @param name the name to set
	 */
	public void setName(String name) {
		this.name = name;
	}

	/**
	 * @return the productoProveedoresId
	 */
	@JsonIgnore
	public Set<ProductoProveedor> getProductoProveedoresId() {
		return productoProveedoresId;
	}

	/**
	 * @param productoProveedoresId the productoProveedoresId to set
	 */
	@JsonIgnore
	public void setProductoProveedoresId(
			Set<ProductoProveedor> productoProveedoresId) {
		this.productoProveedoresId = productoProveedoresId;
	}
	
}
